swag/restswag.py

```diff
@@ -0,0 +1,141 @@
+from pecan import rest
+import inspect
+
+def decorated_controllers():
+    return []
+
+def collect_controllers(RootController):
+    """
+    Traverses project starting with RootController
+    and collects all RestControllers returning a tuple with their respective paths.
+    Concatenates decorated controllers and their paths to the list.
+    :return [(controllers, path),...]
+    """
+    stack = [(RootController, "/")]
+    queue = []
+    while stack:
+        restcontroller = stack.pop()
+        controllers = []
+        for ctrl in dir(restcontroller[0]):
+            if issubclass(type(getattr(restcontroller[0], ctrl)), rest.RestController):
+                controllers.append((getattr(restcontroller[0], ctrl), ctrl))
+        for controller in controllers:
+            sep = '/'
+            if restcontroller[1][-1] == sep: sep = ''
+            stack.append((controller[0], restcontroller[1] + sep + controller[1]))
+        queue.append(restcontroller)
+
+    queue + decorated_controllers()
+    return queue
+
+def collect_methods(controllers):
+    """
+    collects all restful method paths from all restcontrollers in controllers
+    and returns a dictionary with their method-names and methods.
+    :return: {path: method-name: method}
+    """
+    methods = {}
+    for controller in controllers:
+
+        # going through standard pecan methods
+        if getattr(controller[0], 'get_one', None):
+            methodtuple = (controller[1] + "/{" +
+                               getattr(controller[0], 'get_one')._pecan['argspec'][0][1] + "}",
+                           ('get', getattr(controller[0], 'get_one')))
```

Comment:
since we know `getattr(controller[0], 'get_one')` is not empty at this point, due to the `if`, these statements should just use the `controller[0].get_one` instead of getattr(). also for string formatting i think we should use the format function like follows:
```
methodtuple = ('{path}/{{{argspec}}}'.format(path=controller[1], argspec=controller[0].get_one._pecan['argspec'][0][1],
                        ('get', controller[0].get_one))
```